tests/test_config.py

```diff
@@ -386,3 +386,46 @@ def test_umask_config(options, expected):
     with AltArgs(cmdline):
         app = NoConfigApp()
     assert app.cfg.umask == expected
+
+
+@pytest.mark.parametrize("options, expected", [
+    (["--ssl-version", "SSLv23"], 2),
+    (["--ssl-version", "TLSv1"], 3),
+    (["--ssl-version", "2"], 2),
+    (["--ssl-version", "3"], 3),
+])
+def test_ssl_version_named_constants_python3(options, expected):
+    _test_ssl_version(options, expected)
+
+
+@pytest.mark.skipif(sys.version_info < (3, 6),
+    reason="requires python3.6+")
+@pytest.mark.parametrize("options, expected", [
+    (["--ssl-version", "TLS"], 2),
+    (["--ssl-version", "TLSv1_1"], 4),
+    (["--ssl-version", "TLSv1_2"], 5),
+    (["--ssl-version", "TLS_SERVER"], 17),
+])
+def test_ssl_version_named_constants_python36(options, expected):
+    _test_ssl_version(options, expected)
+
+
+@pytest.mark.parametrize("ssl_version", [
+    "FOO",
+    "-99",
+    "99991234"
+])
+def test_ssl_version_bad(ssl_version):
+    c = config.Config()
+    with pytest.raises(ValueError) as excinfo:
```

Comment:
Style nit: This is not a big deal, but we usually prefer `exc` or `err` for this usage in the codebase.